### PR TITLE
Tx commitment

### PR DESCRIPTION
**Pull request by LucasLvy**
# Pull Request type

Please check the type of change your PR introduces:

- [ ] Bugfix
- [x] Feature
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no API changes)
- [ ] Build-related changes
- [ ] Documentation content changes
- [ ] Testing
- [ ] Other (please describe):

# What is the current behavior?

Issue Number: N/A

# What is the new behavior?

- Tx commitment is computed before storing the block.

# Does this introduce a breaking change?

- [ ] Yes
- [x] No

# Other information

### CODE REVIEW

**Comment by LucasLvy** [comment]
Still needs some refacto on the merkle tree part

File: primitives/starknet/src/crypto/hash/mod.rs [hunk 1]
```diff
@@ -38,7 +38,7 @@ pub fn hash_field(hash_type: HashType, x: &FieldElement, y: &FieldElement) -> Re
     // Convert the byte arrays to field elements.
     let x = FieldElement::to_bytes_be(x);
     let y = FieldElement::to_bytes_be(y);
-    // Hash the field elements.
+    // Hasher the field elements.
```
Comment (abdelhamidbakhta) [inline primitives/starknet/src/crypto/hash/mod.rs:41]: bad replacement

File: primitives/starknet/src/crypto/hash/pedersen.rs [hunk 2]
```diff
@@ -29,3 +29,9 @@ impl Hasher for PedersenHasher {
         hash(data)
     }
 }
+
+impl CryptoHasher for PedersenHasher {
+    fn hash(a: FieldElement, b: FieldElement) -> FieldElement {
```
Comment (abdelhamidbakhta) [inline primitives/starknet/src/crypto/hash/pedersen.rs:34]: add missing doc

**Review by abdelhamidbakhta — COMMENTED** [review]
Left some comments. I know the PR is still in draft but I wanted to have a look sooner rather than later.

**Review by abdelhamidbakhta — APPROVED** [review]
lgtm

**Pull request merged**